hummingbot/connector/exchange/litebit_pro/litebit_pro_api_order_book_data_source.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+import asyncio
+import logging
+import time
+import aiohttp
+import pandas as pd
+import hummingbot.connector.exchange.litebit_pro.litebit_pro_constants as constants
+
+from typing import Optional, List, Dict, Any
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.utils.async_utils import safe_gather
+from hummingbot.logger import HummingbotLogger
+from . import litebit_pro_utils
+from .litebit_pro_active_order_tracker import LitebitProActiveOrderTracker
+from .litebit_pro_order_book import LitebitProOrderBook
+from .litebit_pro_websocket import LitebitProWebsocket
+
+
+class LitebitProAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    MAX_RETRIES = 20
+    MESSAGE_TIMEOUT = 30.0
+    SNAPSHOT_TIMEOUT = 10.0
+
+    _logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._logger is None:
+            cls._logger = logging.getLogger(__name__)
+        return cls._logger
+
+    def __init__(self, trading_pairs: List[str] = None):
+        super().__init__(trading_pairs)
+        self._trading_pairs: List[str] = trading_pairs
+        self._snapshot_msg: Dict[str, any] = {}
+
+    @staticmethod
+    async def fetch_trading_pairs() -> List[str]:
+        async with aiohttp.ClientSession() as client:
+            async with client.get(f"{constants.REST_URL}/api/v2/markets", timeout=10) as response:
+                if response.status == 200:
+                    try:
+                        data: List[dict] = await response.json()
+                        return [litebit_pro_utils.convert_from_exchange_trading_pair(item["market"]) for item in
+                                data]
+                    except Exception:
+                        pass
+                        # Do nothing if the request fails -- there will be no autocomplete for kucoin trading pairs
+                return []
+
+    @classmethod
+    async def get_last_traded_prices(cls, trading_pairs: List[str]) -> Dict[str, float]:
+        result = {}
+        async with aiohttp.ClientSession() as client:
+            resp = await client.get(f"{constants.REST_URL}/api/v2/tickers")
```

Comment:
None of the connectors have implemented error handling here. They all seem to return an empty array in case of an error. Do you still think I should add it?